--- meson.build	2020-09-27 01:29:25.000000000 +0200
+++ meson_new.build	2020-12-02 13:49:52.987086190 +0100
@@ -28,11 +28,7 @@
 
 curl_dep = dependency('libcurl')
 
-if build_machine.system() == 'freebsd' or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); }; void main() {}')
-    argplib = cc.find_library('argp')
-else
-    argplib = dependency('', required : false)
-endif
+argplib = cc.find_library('argp')
 
 if get_option('with-openssl') == 'disabled'
     openssl_dep = dependency('', required : false)
