--- meson.build	2020-09-27 01:29:25.000000000 +0200
+++ meson.build.win	2020-12-03 17:26:59.800448587 +0100
@@ -17,16 +17,8 @@
     add_project_arguments('-DFREEBSD', language : 'c')
 endif
 
-if get_option('with-zstd') == 'disabled'
-    zstd_dep = dependency('', required : false)
-else
-    zstd_dep = dependency('libzstd', required : get_option('with-zstd') == 'enabled')
-    if zstd_dep.found()
-        add_project_arguments('-DZCHUNK_ZSTD', language : 'c')
-    endif
-endif
-
-curl_dep = dependency('libcurl')
+zstdlib = cc.find_library('zstd')
+curllib = cc.find_library('curl_a')
 
 if build_machine.system() == 'freebsd' or not cc.links('#include <argp.h>\nstatic error_t parse_opt (int key, char *arg, struct argp_state *state) { argp_usage(state); }; void main() {}')
     argplib = cc.find_library('argp')
